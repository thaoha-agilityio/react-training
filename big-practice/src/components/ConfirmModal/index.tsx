import { memo } from 'react';
import {
  Button,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
  Text,
} from '@chakra-ui/react';

type ConfirmModalProps = {
  isOpen: boolean;
  isLoading?: boolean;
  title: string;
  textCancel: string;
  textSubmit: string;
  text?: string;
  onClose: () => void;
  onSubmit: () => void;
};

const ConfirmModal = ({
  title,
  text,
  textCancel,
  textSubmit,
  isLoading,
  isOpen,
  onSubmit,
  onClose,
}: ConfirmModalProps) => (
  <Modal isOpen={isOpen} onClose={onClose}>
    <ModalOverlay />
    <ModalContent>
      <ModalHeader fontSize={{ base: 'md', md: 'xl' }}>{title}</ModalHeader>
      <ModalCloseButton />
      <ModalBody>
        <Text variant='primary'>{text}</Text>
      </ModalBody>
      <ModalFooter>
        <Button mr={3} onClick={onClose} rounded='xs'>
          {textCancel}
        </Button>
        <Button variant='solid' rounded='xs' onClick={onSubmit} isLoading={isLoading}>
          {textSubmit}
        </Button>
      </ModalFooter>
    </ModalContent>
  </Modal>
);

export default memo(ConfirmModal);
