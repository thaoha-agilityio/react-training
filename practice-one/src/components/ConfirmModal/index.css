.overlay {
  background-color: rgb(0 0 0 / 30%);
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.confirm-modal {
  position: absolute;
  width: 400px;
  height: 180px;
  background: var(--color-secondary);
  z-index: 39;
  font-family: 'Inter';
  border-radius: 8px;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  padding: 20px;
  display: flex;
  flex-direction: column;
}

.close-modal {
  text-align: right;
  cursor: pointer;
}

.modal p {
  text-align: center;
  font-weight: 400;
  font-size: 16px;
}

.button-wrapper {
  display: flex;
  justify-content: space-around;
  margin-top: 50px;
}

.button-wrapper button {
  border: none;
  font-size: 14px;
  border-radius: 5px;
  width: 100px;
  height: 38px;
  font-family: 'Roboto';
  font-weight: 400;
  text-transform: capitalize;
}
