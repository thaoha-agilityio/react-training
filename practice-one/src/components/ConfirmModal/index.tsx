import React from 'react';

// Components
import Button from '../Button';
import closeIcon from '../../assets/images/closeIcon.jpg';

// Css
import './index.css';

interface IProps {
  id: string;

  message: string;
  onClose: () => void;
  onConfirm: (id: string) => void;
}

const ConfirmModal = ({ onClose, id, onConfirm, message }: IProps): React.ReactElement => (
  <div className="overlay">
    <div className="confirm-modal">
      <div className="close-modal">
        <a onClick={onClose}>
          <img src={closeIcon} alt="icon" />
        </a>
      </div>
      <p>{message}</p>
      <div className="button-wrapper">
        <Button variant="secondary" size="small" onClick={onClose}>
          Cancel
        </Button>
        <Button variant="primary" size="small" onClick={() => onConfirm(id)}>
          Save
        </Button>
      </div>
    </div>
  </div>
);

export default ConfirmModal;
