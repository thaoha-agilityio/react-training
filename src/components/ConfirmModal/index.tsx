// Component
import Button from '../Button';

import closeIcon from '../../assets/images/closeIcon.jpg';

// Css
import './index.css';

interface IProps {
  message: string;
  text: string;
  onClose: () => void;
  onConfirm?: () => void;
}

const ConfirmModal = ({ onClose, onConfirm, message, text }: IProps): React.ReactElement => (
  <div className="overlay">
    <div className="confirm-modal">
      <div className="close-modal">
        <a onClick={onClose}>
          <img src={closeIcon} alt="icon" />
        </a>
      </div>
      <p>{message}</p>
      <div className="button-wrapper">
        <Button variant="secondary" size="small" onClick={onClose}>
          Cancel
        </Button>
        <Button
          variant={text === 'delete' ? 'danger' : 'primary'}
          size="small"
          onClick={() => onConfirm?.()}
        >
          {text}
        </Button>
      </div>
    </div>
  </div>
);

export default ConfirmModal;
