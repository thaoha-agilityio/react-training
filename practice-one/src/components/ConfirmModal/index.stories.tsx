import { ComponentMeta, ComponentStory } from '@storybook/react';

import ConfirmModal from '.';

export default {
  title: 'ConfirmModal',
  component: ConfirmModal,
} as ComponentMeta<typeof ConfirmModal>;

const Template: ComponentStory<typeof ConfirmModal> = (args) => <ConfirmModal {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  message: 'Do you wanna delete the item?',
};
