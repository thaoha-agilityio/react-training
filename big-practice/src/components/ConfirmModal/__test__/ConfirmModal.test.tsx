import { fireEvent, render } from '@testing-library/react';

import ConfirmModal from '..';

const mockProps = {
  isOpen: true,
  title: 'Delete Confirmation',
  textCancel: 'Cancel',
  textSubmit: 'Yes, Delete',
  text: 'Are you sure you want to delete this item?',
  onClose: jest.fn(),
  onSubmit: jest.fn(),
  isLoading: false,
};

describe('ConfirmModal Component', () => {
  it('renders product information correctly', () => {
    const { getByText } = render(<ConfirmModal {...mockProps} />);

    expect(getByText(mockProps.title)).toBeInTheDocument();
    expect(getByText(mockProps.text)).toBeInTheDocument();

    const cancelBtn = getByText(mockProps.textCancel);
    fireEvent.click(cancelBtn);
    expect(mockProps.onClose).toHaveBeenCalled();

    const submitBtn = getByText(mockProps.textSubmit);
    fireEvent.click(submitBtn);
    expect(mockProps.onSubmit).toHaveBeenCalled();
  });
});
