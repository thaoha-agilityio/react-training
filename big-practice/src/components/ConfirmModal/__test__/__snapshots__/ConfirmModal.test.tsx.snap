// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal Component renders product information correctly 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;
